Python, Ruby and Perl integration builds consistently fail to install semantic release dependency

I'm getting various `'semantic-release/last-release-npm' is not in the npm registry` errors when installing my project on Travis CI, not sure exactly why. The name of the library after `/` is different every time, but it's always prefixed `semantic-release/`.

It happens only in [Ruby](https://travis-ci.org/apiaryio/dredd/builds/152406442), [Perl](https://travis-ci.org/apiaryio/dredd/builds/152406477) and [Python](https://travis-ci.org/apiaryio/dredd/builds/152406424) environment and it happens (= fails) consistently. As I said, name of the library changes with every build. I have very similar builds with [Go](https://travis-ci.org/apiaryio/dredd/builds/152406522) or [PHP](https://travis-ci.org/apiaryio/dredd/builds/152406495) and they install correctly.

```
npm ERR! 404 Not Found
npm ERR! 404 
npm ERR! 404 'semantic-release/last-release-npm' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it
npm ERR! 404 It was specified as a dependency of 'semantic-release'
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, or http url, or git url.
npm ERR! System Linux 3.13.0-40-generic
npm ERR! command "/home/travis/.nvm/v0.10.36/bin/node" "/home/travis/.nvm/v0.10.36/bin/npm" "link" "--python=python2" "./__dredd__"
npm ERR! cwd /home/travis/build/apiaryio/dredd
npm ERR! node -v v0.10.36
npm ERR! npm -v 1.4.28
npm ERR! code E404
npm ERR! not ok code 0
```

I'm quite desperate, not sure where to look for the culprit - whether it's something with the semantic release package or whether it's in Dredd's code, Travis CI, ... 😞 

-------------

I described the problem in detail also in https://github.com/semantic-release/semantic-release/issues/273.

--- a/scripts/test-hooks-handlers.coffee
+++ b/scripts/test-hooks-handlers.coffee
@@ -201,13 +201,12 @@ prependToTravisConfigProperty = (config, property, value) ->
 # *  Removes 'deploy'
 # *  Adds status reporting
 adjustTravisBuildConfig = (pullRequestId, testedCommit, jobName, matrixName) ->
-  if pullRequestId
-    # We will want to report under the Pull Request, so we will need GitHub
-    # token present in the configuration.
-    execSync("""\
-      travis encrypt GH_TOKEN=#{process.env.GH_TOKEN} \
-        --add --append --no-interactive --repo=apiaryio/dredd #{DROP_OUTPUT}
-    """)
+  # We will want to report under the Pull Request, so we will need GitHub
+  # token present in the configuration.
+  execSync("""\
+    travis encrypt GH_TOKEN=#{process.env.GH_TOKEN} \
+      --add --append --no-interactive --repo=apiaryio/dredd #{DROP_OUTPUT}
+  """)
 
   # Read contents of the '.travis.yml' file
   contents = fs.readFileSync(TRAVIS_CONFIG_FILE)
@@ -221,37 +220,36 @@ adjustTravisBuildConfig = (pullRequestId, testedCommit, jobName, matrixName) ->
   # Remove any deploy configuration, just to be sure
   delete config.deploy
 
-  if pullRequestId
-    # Enhance the build configuration so it reports results back to PR and deletes
-    # the branch afterwards.
-    command = createStatusCommand(testedCommit,
-      state: 'pending'
-      description: 'Dependent build created'
-      context: "continuous-integration/travis-ci/#{jobName}"
-      target_url: "https://travis-ci.org/apiaryio/dredd/builds/$TRAVIS_BUILD_ID"
-    )
-    prependToTravisConfigProperty(config, 'before_install', command)
+  # Enhance the build configuration so it reports results back to PR and deletes
+  # the branch afterwards.
+  command = createStatusCommand(testedCommit,
+    state: 'pending'
+    description: 'Dependent build created'
+    context: "continuous-integration/travis-ci/#{jobName}"
+    target_url: "https://travis-ci.org/apiaryio/dredd/builds/$TRAVIS_BUILD_ID"
+  )
+  prependToTravisConfigProperty(config, 'before_install', command)
 
-    command = createStatusCommand(testedCommit,
-      state: 'error'
-      description: 'Dependent build finished'
-      context: "continuous-integration/travis-ci/#{jobName}"
-      target_url: "https://travis-ci.org/apiaryio/dredd/builds/$TRAVIS_BUILD_ID"
-    )
-    prependToTravisConfigProperty(config, 'after_failure', command)
+  command = createStatusCommand(testedCommit,
+    state: 'error'
+    description: 'Dependent build finished'
+    context: "continuous-integration/travis-ci/#{jobName}"
+    target_url: "https://travis-ci.org/apiaryio/dredd/builds/$TRAVIS_BUILD_ID"
+  )
+  prependToTravisConfigProperty(config, 'after_failure', command)
 
-    command = createStatusCommand(testedCommit,
-      state: 'success'
-      description: 'Dependent build finished'
-      context: "continuous-integration/travis-ci/#{jobName}"
-      target_url: "https://travis-ci.org/apiaryio/dredd/builds/$TRAVIS_BUILD_ID"
-    )
-    prependToTravisConfigProperty(config, 'after_success', command)
+  command = createStatusCommand(testedCommit,
+    state: 'success'
+    description: 'Dependent build finished'
+    context: "continuous-integration/travis-ci/#{jobName}"
+    target_url: "https://travis-ci.org/apiaryio/dredd/builds/$TRAVIS_BUILD_ID"
+  )
+  prependToTravisConfigProperty(config, 'after_success', command)
 
-    config.after_success.push('if [[ $TRAVIS_BRANCH = master ]]; then echo "Deleting aborted (master)" && exit 1; fi')
-    config.after_success.push('git branch -D $TRAVIS_BRANCH')
-    config.after_success.push("git remote set-url origin \"https://$GH_TOKEN@github.com/apiaryio/dredd.git\" #{DROP_OUTPUT}")
-    config.after_success.push("git push origin -f --delete $TRAVIS_BRANCH #{DROP_OUTPUT}")
+  config.after_success.push('if [[ $TRAVIS_BRANCH = master ]]; then echo "Deleting aborted (master)" && exit 1; fi')
+  config.after_success.push('git branch -D $TRAVIS_BRANCH')
+  config.after_success.push("git remote set-url origin \"https://$GH_TOKEN@github.com/apiaryio/dredd.git\" #{DROP_OUTPUT}")
+  config.after_success.push("git push origin -f --delete $TRAVIS_BRANCH #{DROP_OUTPUT}")
 
   # Save all changes
   fs.writeFileSync(TRAVIS_CONFIG_FILE, yaml.dump(config), 'utf-8')
@@ -330,8 +328,7 @@ ensureGitOrigin()
 
 
 JOBS.forEach(({name, repo, matrixName}) ->
-  id = if pullRequestId then "pr#{pullRequestId}/#{buildId}" else buildId
-  integrationBranch = "dependent-build/#{id}/#{name}"
+  integrationBranch = "dependent-build/pr#{pullRequestId}/#{buildId}/#{name}"
   integrationBranches.push(integrationBranch)
   console.log("Preparing branch #{integrationBranch}")
 

--- a/scripts/test-hooks-handlers.coffee
+++ b/scripts/test-hooks-handlers.coffee
@@ -87,7 +87,7 @@ JOBS = [
 
 TRAVIS_CONFIG_FILE = '.travis.yml'
 LINKED_DREDD_DIR = './__dredd__'
-RE_DREDD_INSTALL_CMD = /npm ([ \-=\w]+ )?i(nstall)? ([ \-=\w]+ )?dredd/
+RE_DREDD_INSTALL_CMD = /npm ([ \-=\w]+ )?i(nstall)? ([ \-=\w]+ )?dredd(@\w+)?/
 DREDD_LINK_CMD = "npm link --python=python2 #{LINKED_DREDD_DIR}"
 
 
@@ -125,7 +125,8 @@ buildFindExcludes = (excludedPaths) ->
   return expressions.join(' -and ')
 
 
-# Replaces given pattern with replacement in given file. Returns boolean whether
+# Replaces command installing Dredd from npm with one installing Dredd from
+# local folder with source code of the tested version. Returns boolean whether
 # any changes were made.
 replaceDreddInstallation = ->
   contents = fs.readFileSync(TRAVIS_CONFIG_FILE, 'utf-8')
@@ -219,6 +220,9 @@ adjustTravisBuildConfig = (pullRequestId, testedCommit, jobName, matrixName) ->
 
   # Remove any deploy configuration, just to be sure
   delete config.deploy
+
+  # Make sure the latest npm gets installed
+  prependToTravisConfigProperty(config, 'before_install', 'npm install -g npm@latest')
 
   # Enhance the build configuration so it reports results back to PR and deletes
   # the branch afterwards.

